router/router.go

```
@@ -101,6 +105,19 @@ func (a *App) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	// Start the server in a goroutine
+	go func() {
+		if err := http.ListenAndServe(addr, a); err != nil {
+			log.Fatalf("Serve failed to start: %v", err)
+		}
+	}()
+
+	// Wait for interrupt signal
+	<-stop
+
+	return
```

golangci-lint

S1023: redundant `return` statement

(gosimple)